starrocks-stream-load-sdk/src/main/java/com/starrocks/data/load/stream/compress/LZ4FrameCompressionCodec.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2021-present StarRocks, Inc. All rights reserved.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.starrocks.data.load.stream.compress;
+
+import net.jpountz.lz4.LZ4Compressor;
+import net.jpountz.lz4.LZ4Factory;
+import net.jpountz.lz4.LZ4FrameOutputStream;
+import net.jpountz.xxhash.XXHash32;
+import net.jpountz.xxhash.XXHashFactory;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Map;
+
+/** Compress data using LZ4_FRAME. */
+public class LZ4FrameCompressionCodec implements CompressionCodec {
+
+    public static final String NAME = "LZ4_FRAME";
```

Comment:
is LZ4 compatible？

Reply:
We need streaming compression here, so use lz4_frame. And SR also does not support lz4, see https://github.com/StarRocks/starrocks/pull/43732